packages/site/src/layout/ComponentView.tsx

```diff
@@ -44,6 +44,11 @@ export const ComponentView = () => {
     };
   }, []);
 
+  useEffect(() => {
+    // Reset tab to 0 (or the first available tab) when the component name changes
+    setTab(0);
+  }, [name]);
```

Comment:
This fixes the issue where if you're on a component page with an Implement tab, then go to a component without an Implement tab, you land on a component page that shows no content.
This also happens when moving from components with a Mobile tab and ones without - it's just a bit awkward.

https://github.com/user-attachments/assets/8aa59dbc-561f-4a30-b6ff-9b505192ac16